sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -2053,9 +2063,74 @@ class Analyzer(override val catalogManager: CatalogManager)
                   name, other.getClass.getCanonicalName)
               }
             }
-          case u @ UnresolvedFunction(funcId, arguments, isDistinct, filter, ignoreNulls) =>
+          case UnresolvedFunction(NonSessionCatalogAndIdentifier(v2Catalog, ident), arguments,
```

Comment:
This logic is not correct - when there is no namespace part in the identifier, it should check built-in and temp functions first, and also handle function references in views. Only if this part failed it should use the default catalog (whether that be the V1 session catalog or a V2 catalog). Will fix this.